src/key.rs

```diff
@@ -291,6 +304,22 @@ impl PublicKey {
         ret
     }
 
+    #[inline]
+    /// Negates the pk to the pk `self` in place
+    /// Will return an error if the pk would be invalid.
+    pub fn negate_assign<C: Verification>(
+        &mut self,
+        secp: &Secp256k1<C>
+    ) -> Result<(), Error> {
+        unsafe {
+            if ffi::secp256k1_ec_pubkey_negate(secp.ctx, &mut self.0 as *mut _) == 1 {
+                Ok(())
+            } else {
+                Err(Error::InvalidPublicKey)
```

Comment:
Replace this with `assert_eq(res, 1);`, as the header defines it should always return 1
https://github.com/rust-bitcoin/rust-secp256k1/blob/master/secp256k1-sys/depend/secp256k1/include/secp256k1.h#L574